src/services/expenseService.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+const { models } = require('../models/models');
+
+class ExpenseService {
+  async createExpense(userId, spentAt, title, amount, category, note) {
+    const newExpense = await models.Expense.create({
+      userId,
+      spentAt,
+      title,
+      amount,
+      category,
+      note,
+    });
+
+    return newExpense;
+  }
+
+  async getAllExpenses({ userId, category }) {
```

Comment:
Intead of `({ userId, category })`, you can use `(where = {})`
Ignore the ifs and them: `const expenses = await models.Expense.findAll({ where })`